enhancements/security/tls-artifacts-registry.md

```diff
@@ -0,0 +1,308 @@
+
+### Goals
+
+Create a registry of all managed TLS artifacts, and ensure that no new 
+certificates are added without sufficient metadata.
+Add a test to the conformance suite that verifies the actual cluster TLS 
+artifacts have necessary metadata.
+
+### Non-Goals
+
+## Proposal
+
+* Define a structure for the certificate registry origin repo
+  * Each TLS artifact can be either
+    * on-disk file
+    * in-cluster secret or config map
+  * Additional expected metadata stored for this artifact
+    * JIRA component
+    * TLS artifact description (one sentence describing the purpose 
+      of this artifact)
+  * Each TLS artifact is associated with a set of required metadata
+* Create a test which saves existing information about TLS artifacts 
+  in cluster.
+  * Some secrets may be versioned, contain hash or IP address / DNS 
+    address - i.e. 
+    `etcd-peer-ip-10-10-10-10.us-east-1.internal` becomes 
+    `etcd-peer-<master-2>` etc.
+    The test would not store versioned copies and replace IPs with "master-n"
+    to make them look uniform when saving raw TLS information
+* Create a new test, which ensures that all secrets are present in the 
+  registry
+* Create a test that verifies that TLS artifacts in a cluster have 
+  necessary annotations to provide required metadata.
+* Some TLS artifacts can be added to known violation list so that the test 
+  is not permafailing to give teams time to update metadata. Each metadata 
+  property has its own violation list.
+* Metadata is split into required and optional properties. The test would 
+  record missing optional metadata and fail if required metadata is missing.
+  This gives time to update certificate metadata when optional metadata 
+  requirements become required. Violation files are meant to be "remove-only" 
+  mode, meaning when metadata property becomes required users are not meant 
+  to be added to the respective violation file.
+* Initially required metadata properties would be "owning JIRA component" and 
+  "one line description of artifact purpose". Later on other properties would be 
+  added and eventually become required for new certificates.
+* TLS artifact registry is being composed as a JSON file from several 
+  parts (usually generated from the test job artifacts).
+* In the `origin` repo create `hack/update-tls-artifacts.json` and 
+  `hack/verify-tls-artifacts.json` scripts to compose TLS artifact 
+  registry JSON from several JSONs and verify that all necessary 
+  metadata is set respectively. `update` script also generate violation lists
+  in JSON and Markdown formats. For each required metadata field the Markdown 
+  report can be customized.
+* Update library-go functions to set metadata when the TLS artifact is 
+  being generated, so that most operators would use the recommended 
+  method.
+
+### Workflow Description
+
+#### Adding a new TLS artifact
+
+The cluster administrator opens the TLS artifacts registry in the `origin` repo 
```

Comment:
Have you consider publishing the registry in our docs?  Would it be feasible?

Reply:
Or at least a link from the docs describing this problem to a particular file in origin repo.

Reply:
We already have docs generated in openshift/api about which certificates we expect and how they interact. This enhancement is more focused on ensuring that certs in cluster actually match this model. These features definitely overlap, but there is no need to publish actual certs from CI cluster to docs.

Its worth adding a link to the api docs though, its useful to keep expected and actual state in sync